File: Test_mobile/Test_mobile/Entity/Lesson.swift
struct Lesson: Codable {
    var companyID: String
    var createdAt: Int
    var name: String
    var id: String
    var status: String
    var summary: Summary
    var removeLessonDelegate: RemovableLesson?

    private enum CodingKeys: String, CodingKey {
        case name, id, status, summary
        case companyID = "company_id"
        case createdAt = "created_at"
    }
}

File: lib/src/model/lesson.dart
import 'package:teste_mobile/src/model/summary.dart';

class Lesson {
  int companyId;
  DateTime createdAt;
  String name;
  String id;
  String status;
  Summary summary;

  Lesson.fromJson(Map<String, dynamic> json)
      : companyId = json['company_id'],
        createdAt = json['created_at'],
        name = json['name'],
        id = json['id'],
        status = json['status'],
        summary = Summary.fromJson(json['summary']);
}
